### ISSUE TITLE
Effect Shader does *NOT* work anymore

### ISSUE THREAD

**Issue by elmkni** [castle-engine/view3dscene#24]
Hi Michalis,

it's me, Elmar (again) ...

I've just tested the latest snapshot of View3DScene.exe

`view3dscene-3.18.0-win64-x86_64.zip | 2020-03-10 06:09 | 4.1M | Latest Windows version (64-bit)`

and got following complaint by view3dscene when I use an 'Effect' shader:

`VRML/X3D: Plug name "texture_apply" not declared (in shader type "Fragment")`

although this file worked perfectly with an older view3dscene snapshot from 2020-02-20!

it should look like this:
![BackFog_animated fx](https://user-images.githubusercontent.com/53652478/76335649-2544d900-62f5-11ea-987f-a610bcb9e2fb.jpg)

It does work when I use a complete 'ComposedShader' --- am I missing something?

I'll attach the test files (both: Effect shader and ComposedShader version) for you to test.

Please help!

With best regards,

Elmar

[view3dscene_issue_24.zip](https://github.com/castle-engine/view3dscene/files/4313632/view3dscene_issue_24.zip)

**Comment by michaliskambi** [comment]
The plug name changed to `main_texture_apply`. Unfortunately I had to change it, the previous name was really bad (it does *not* affect all textures). I didn't yet announce this change, but it's related to recent PBR and material/lighting improvements on https://castle-engine.io/wp/2020/03/07/physically-based-rendering-and-lots-of-other-lighting-materials-upgrades-in-castle-game-engine-and-x3d-4-0/ .

**Comment by elmkni** [comment]
Hi Michalis,

thanks for the quick answer!

So I have to change all my files ...

with best regards,

Elmar

**Comment by michaliskambi** [comment]
Hm, let me implement an upgrade path actually. I can make the `texture_apply` working again, just deprecated. This should be easy, I'll do it ASAP.

**Comment by michaliskambi** [comment]
I pushed a simple fix to enable PLUG name `texture_apply` for backward compatibility.

In about 15 minutes, latest view3dscene snapshot from http://michalis.ii.uni.wroc.pl/view3dscene-snapshots/ should have it working :)

**Comment by elmkni** [comment]
Hi Michalis,

thanks a lot for the 'PLUG_main_texture_apply' work-around!

In the meantime I had already updated my effect files ;)
but it is always good to have backward compability!

With best regards,

Elmar

P.S.: 
I've checked your test-suite concerning the new X3D v4.0 stuff (especially PBR and unlit material), and it looks pretty awesome! I am really looking forward to all the possibilities X3D v4.0 will bring!
